src/Parser/Lexer/Common.idr

```diff
@@ -45,3 +43,26 @@ ident : Flavour -> Lexer
 ident flavour =
   (pred $ isIdentStart flavour) <+>
     (many . pred $ isIdentTrailing flavour)
+
+export
+isIdentNormal : String -> Bool
+isIdentNormal = isIdent Normal
+
+export
+identNormal : Lexer
+identNormal = ident Normal
+
+export
+identAllowDashes : Lexer
+identAllowDashes = ident AllowDashes
+
+namespaceIdent : Lexer
+namespaceIdent = ident Capitalised <+> many (is '.' <+> ident Capitalised <+> expect (is '.'))
+
+export
+namespacedIdent : Lexer
+namespacedIdent = namespaceIdent <+> opt (is '.' <+> identNormal)
```

Comment:
Oh, nice!

I think it even keeps the accuracy that you originally wanted (my suggestion wouldn't).

Reply:
Perhaps `namespaceIdent` could be called just `namespace`?

Reply:
Impossible since namespace is a keyword in Idris2